thunder/tests/test_quantization.py

```diff
@@ -0,0 +1,81 @@
+import torch
+import time
+from thunder.transforms.quantization import BitsAndBytesLinearQuant4bit
+
+
+def test_cpu_quantization():
+    # Initialize quantization transform
+    quant_transform = BitsAndBytesLinearQuant4bit()
+
+    # Create a tensor on CPU
+    weight = torch.randn(3, 3, device="cpu")
+
+    # Quantize weight (expect only the quantized tensor, not a tuple)
+    quantized_weight = quant_transform.quantize_weight(weight)
+
+    # Check that the quantized tensor has fewer or equal elements due to compression
+    original_num_elements = weight.numel()
+    quantized_num_elements = quantized_weight.numel()
+
+    assert quantized_weight is not None, "Quantized weight is None"
+    assert (
+        quantized_num_elements <= original_num_elements
+    ), "Quantized tensor should have fewer or equal elements due to compression"
+
+
+def test_gpu_quantization():
+    if not torch.cuda.is_available():
+        return
+
+    # Initialize quantization transform
+    quant_transform = BitsAndBytesLinearQuant4bit()
+
+    # Create a tensor on GPU
+    weight = torch.randn(3, 3, device="cuda")
+
+    # Quantize weight (expect only the quantized tensor, not a tuple)
+    quantized_weight = quant_transform.quantize_weight(weight)[0]
+
+    # Check that the quantized tensor has fewer or equal elements due to compression
+    original_num_elements = weight.numel()
+    quantized_num_elements = quantized_weight.numel()
+
+    assert quantized_weight is not None, "Quantized weight is None"
+    assert (
+        quantized_num_elements <= original_num_elements
+    ), "Quantized tensor should have fewer or equal elements due to compression"
+
+
+# Optional: Performance tests
```

Comment:
Hey, I think these are cool to demonstrate the value, but tests are maybe not the best place for it.
If we want to show the performance, maybe we could create a quantization notebook or so? Similar to the "what's going on under the hood in fsdp".